src/xunit.netcore.extensions/Discoverers/ConditionalTestDiscoverer.cs

```diff
@@ -48,16 +48,36 @@ internal class ConditionalTestDiscoverer
                     continue;
                 }
 
-                string[] symbols = conditionMemberName.Split('.');
                 Type declaringType = testMethodDeclaringType;
 
-                if (symbols.Length == 2)
+                // We have qualified type name with the assembly name, something like
+                // [ConditionalFact("System.PlatformDetection, CoreFx.Private.TestUtilities!" + nameof(PlatformDetection.IsNonZeroLowerBoundArraySupported))]
+                // We don't use '.' as separator in such case and use '!' because the qualified type name can have '.' to include the type namespace.
+                if (conditionMemberName.IndexOf('!') > 0)
                 {
-                    conditionMemberName = symbols[1];
-                    ITypeInfo type = testMethod.TestClass.Class.Assembly.GetTypes(false).Where(t => t.Name.Contains(symbols[0])).FirstOrDefault();
-                    if (type != null)
+                    // get the method name
+                    string[] symbols = conditionMemberName.Split('!');
+                    if (symbols.Length == 2)
```

Comment:
Well, if the length is 1 it should keep doing what it was doing... there's no reason we should change the attributes that are targeting local, functionality-named members.

```C#

public static bool PlatformSupportsRsaPss { get; } = PlatformDetection.IsWindows10;

[ConditionalFact(nameof(PlatformSupportsRsaPss)]
public static void SomePssTest()
{
    ...
}
```

That's my preferred style, since it says why it's conditional, not where it's conditional.  (So, to me, the need for change here doesn't really exist)